examples/nlp/semantic_similarity_with_keras_nlp.py

```diff
@@ -0,0 +1,292 @@
+
+keras_nlp.models.BertClassifier class attaches classification head to the BERT Backbone, 
+mapping  the backbone outputs to logit output suitable for a classification task. This 
+significantly reduces need of custom code.
+
+Here we'll use this model with pre-trained weights. `from_preset()` method allows you
+to use your own preprocessor. Here we'll set the `num_classes` as 3 for SNLI dataset
+"""
+
+bert_classifier = keras_nlp.models.BertClassifier.from_preset(
+    "bert_tiny_en_uncased", num_classes=3
+)
+
+"""
+Take a note that BERT tiny has 4,386,307 trainable parameters. KerasNLP task models come with
+compilation defaults. Let's train the model we just instantiated, by calling the fit() method with 
+"""
+
+bert_classifier.fit(train_ds, validation_data=val_ds, epochs=1)
+
+"""
+
+Our BERT classifier gave us ~65% accuracy on validation split, let's see how it performs
+on Test split
+
+### Evaluate the performance of the trained model on test data.
+"""
+bert_classifier.evaluate(test_ds)
+
+"""
+Our baseline bert gave almost similar (~68%) accuracy on the test split. Let's see if we can
+improve it.
+
+Let's recompile our model with a different learning rate and see performance
+"""
+bert_classifier = keras_nlp.models.BertClassifier.from_preset(
+    "bert_tiny_en_uncased", num_classes=3
+)
+
+bert_classifier.compile(
+    loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+    optimizer=tf.keras.optimizers.Adam(1e-4),
+    metrics=["accuracy"],
+)
+
+bert_classifier.fit(train_ds, validation_data=val_ds, epochs=1)
+
+bert_classifier.evaluate(test_ds)
+"""
+This time we got ~72% validation accuracy on val and test split. Not bad for 1 epoch ! Let's save our model
+for now and learn how to perform inference with it. We took batch size of 512 to utilize our GPUs fully.
+
+Let's see if we can improve it further. Let's use a learning rate scheduler this time.
+"""
+
+
+class TriangularSchedule(keras.optimizers.schedules.LearningRateSchedule):
+    """Linear ramp up for `warmup` steps, then linear decay to zero at `total` steps."""
+
+    def __init__(self, rate, warmup, total):
+        self.rate = tf.cast(rate, dtype="float32")
+        self.warmup = tf.cast(warmup, dtype="float32")
+        self.total = tf.cast(total, dtype="float32")
+
+    def __call__(self, step):
+        step = tf.cast(step, dtype="float32")
+        multiplier = tf.cond(
+            step < self.warmup,
+            lambda: step / self.warmup,
+            lambda: (self.total - step) / (self.total - self.warmup),
+        )
+        return tf.maximum(self.rate * multiplier, 0.0)
+
+
+bert_classifier = keras_nlp.models.BertClassifier.from_preset(
+    "bert_tiny_en_uncased", num_classes=3
+)
+
+bert_classifier.compile(
+    loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+    optimizer=tf.keras.optimizers.experimental.AdamW(
+        TriangularSchedule(0.001, 1000, 5000)
```

Comment:
Maybe we should show how to get an accurate count of the train steps here. It's a bit tricky, but useful to show to users. Something like this (only first two lines are new)...

```python
# Get the total count of training batches.
# This requires walking the dataset to filter all -1 labels.
total_steps = sum(1 for _ in train_ds.as_numpy_iterator())
warmup_steps = int(total_steps * 0.2)

bert_classifier.compile(
    loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
    optimizer=tf.keras.optimizers.experimental.AdamW(
        TriangularSchedule(0.001, warmup_steps, total_steps)
    ),
    metrics=["accuracy"],
)
```

Reply:
Hey @mattdangerw ! cool !
but please take a look at following:

![image](https://user-images.githubusercontent.com/51750587/227028150-25da5a5a-92e6-4c8e-964e-208eff221383.png)

Reply:
It saturates.

Reply:
Ah, I might have made a mistake here, I think we want to do this rate across the total number of train steps for *all epochs*. If you update to this, how do things look?

```python
epochs = 3
# Get the total count of training batches.
# This requires walking the dataset to filter all -1 labels.
total_steps = sum(1 for _ in train_ds.as_numpy_iterator()) * epochs
warmup_steps = int(total_steps * 0.2)
```